sources/packages/backend/apps/api/src/route-controllers/overaward/_tests_/e2e/overaward.aest.controller.getOverawardBalance.e2e-spec.ts

```diff
@@ -0,0 +1,132 @@
+  let app: INestApplication;
+  let studentRepo: Repository<Student>;
+  let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    studentRepo = dataSource.getRepository(Student);
+    disbursementOverawardRepo = dataSource.getRepository(DisbursementOveraward);
+  });
+
+  it("Should get the overaward balance when AEST user belongs to any of the allowed groups", async () => {
+    // Arrange
+    const student = await studentRepo.save(createFakeStudent());
+    await disbursementOverawardRepo.save(
+      createFakeDisbursementOveraward({ student }),
+    );
+    const endpoint = `/aest/overaward/student/${student.id}/balance`;
+    const expectedPermissions = [
+      {
+        aestGroup: AESTGroups.BusinessAdministrators,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: AESTGroups.Operations,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: AESTGroups.OperationsAdministrators,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+      {
+        aestGroup: AESTGroups.MOFOperations,
+        expectedHttpStatus: HttpStatus.OK,
+      },
+    ];
+
+    // Act/Assert
+    for (const permission of expectedPermissions) {
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(await getAESTToken(permission.aestGroup), BEARER_AUTH_TYPE)
+        .expect(permission.expectedHttpStatus);
+    }
+  });
+
+  it("Should return correct value for overaward balance", async () => {
+    // Arrange
+    const student = await studentRepo.save(createFakeStudent());
+    // Create an overaward.
+    const legacyOveraward = createFakeDisbursementOveraward({ student });
+    legacyOveraward.disbursementValueCode = "CSLF";
+    legacyOveraward.overawardValue = 500;
+    legacyOveraward.originType =
+      DisbursementOverawardOriginType.LegacyOveraward;
+    // Create a manual overaward deduction.
+    const manualOveraward = createFakeDisbursementOveraward({ student });
+    manualOveraward.disbursementValueCode = "CSLF";
+    manualOveraward.overawardValue = -100;
+    manualOveraward.originType = DisbursementOverawardOriginType.ManualRecord;
+    // Persist the overawards.
+    await disbursementOverawardRepo.save([legacyOveraward, manualOveraward]);
+    const endpoint = `/aest/overaward/student/${student.id}/balance`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(
+        await getAESTToken(AESTGroups.BusinessAdministrators),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.OK)
+      .then((response) => {
+        expect(response.body.overawardBalanceValues.CSLF).toStrictEqual(
+          (400).toFixed(2),
+        );
+      });
+  });
+
+  it("Should return an empty response when invalid student id is provided", async () => {
```

Comment:
If the student does not exist it should return a not found, once the student exists and there is no data, we will return an empty result. Others API endpoints for AEST are using the below check.
```ts
const student = await this.studentService.getStudentById(studentId);
if (!student) {
  throw new NotFoundException("Student not found.");
}
```

Reply:
I was following the pattern of API returning a collection of data where we are not used to validating the path param values against the database. 

For e.g. something like this. 
![image](https://user-images.githubusercontent.com/54600590/219174742-299f4d53-f48c-4606-9d2a-9c2556ec6864.png)

Reply:
I believe that we have a mixed pattern then, as shown below we have other endpoints doing the check where some of them are also returning a collection. As we talked it seems more accurate to return the `NotFoundException` when the student was not found and the empty collection when the student was found but there is no data.

@Get("student/:studentId")
@Post("student/:studentId")
@Patch(":studentId/save-uploaded-files")
@Get(":studentId/sin-validations")
@Post(":studentId/sin-validations")
@Patch(":studentId/sin-validations/:sinValidationId")
@Patch(":studentId/sin-validations/:sinValidationId")
@Post(":studentId/files")
@Get(":studentId/application-summary")